src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -157,18 +161,48 @@ private void VerifySigning(string nupkgPath)
                 return;
             }
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (!_isNuGetTool)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
+                        out string commandOutput))
+                    {
+                        throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
+                                                                 Environment.NewLine +
+                                                                 commandOutput);
+                    }
+                }
+            }
+            else
             {
-                if (!_firstPartyNuGetPackageSigningVerifier.Verify(new FilePath(nupkgPath),
-                    out string commandOutput))
+                if(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 {
-                    throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
-                                                             Environment.NewLine +
-                                                             commandOutput);
+                    if (!NuGetVerify(new FilePath(nupkgPath), out string commandOutput))
+                    {
+                        throw new NuGetPackageInstallerException(LocalizableStrings.FailedToValidatePackageSigning +
+                                                                 Environment.NewLine +
+                                                                 commandOutput);
+                    }
                 }
+                else
+                {
+                    _reporter.WriteLine(LocalizableStrings.NuGetPackageSignatureVerificationSkipped);
+                }   
             }
         }
 
+        private static bool NuGetVerify(FilePath nupkgToVerify, out string commandOutput)
+        {
+            var args = new[] { "verify", "--all", nupkgToVerify.Value };
+            var command = new DotNetCommandFactory(alwaysRunOutOfProc: true)
+                .Create("nuget", args);
```

Comment:
Did you get feedback from the NuGet team on how to do the signature verification? Is starting a new process the only way to do this?  Are there NuGet APIs we could call in-process instead?

Reply:
Thank you very much. I talked to @Damon and he suggests that calling the [nuget verify command](https://learn.microsoft.com/dotnet/core/tools/dotnet-nuget-verify) is the preferred approach because it is easier to maintain over the long term. "It would be possible to invoke the same functionality via API, but any API surface that's exposed today was less the result of an intentional, well crafted API design and more the result of a "let's make everything public by default" team policy. " Overall, it seems that calling the nuget verify command is valid and is a good approach.